### PR TITLE
Add service discovery section to features docs

### PR DESCRIPTION
**Pull request by awh**

### CODE REVIEW

File: site/features.md [hunk 1]
```diff
@@ -281,6 +282,20 @@ fully-qualified domain name:
...
+are launched:
+
+    host1$ docker run -dti --name=service ubuntu
+    host1$ docker run -ti  --name=client ubuntu
+    root@client:/# ping service
+
+The service discovery mechanism supports hot swapping, load balancing
+and fault resilience; see the [weaveDNS](weavedns.html) documentation
+for more details.
+
```
Comment (rade) [inline site/features.md]: Call this "Naming and discovery", reword slightly, and move it right after the IPAM section. And don't give the client a name.

**Comment by awh** [comment]
Addressed comments.

File: site/features.md [hunk 2]
```diff
@@ -108,6 +109,19 @@ IP addresses of external services the hosts or containers need to
 connect to. The same IP range must be used everywhere, and the
 individual IP addresses must, of course, be unique.
 
+### <a name="naming-and-discovery"></a>Naming and discovery
+
+Application containers automatically register themselves for discovery
+simply by giving them a container name when they are launched:
```
Comment (rade) [inline site/features.md]: "Named containers are automatically registered in weaveDNS(link), which makes them discoverable through simple name lookups:"

File: site/features.md [hunk 3]
```diff
@@ -108,6 +109,19 @@ IP addresses of external services the hosts or containers need to
...
+
+    host1$ docker run -dti --name=service ubuntu
+    host1$ docker run -ti ubuntu
+    root@7b21498fb103:/# ping service
+
+This feature supports hot swapping, load balancing and fault
+resilience; see the [weaveDNS](weavedns.html) documentation for more
```
Comment (rade) [inline site/features.md]: "...supports load balancing, fault resilience and hot swapping" (same order as in weaveDNS docs)

**Pull request merged**